workflow/util/util.go

```diff
@@ -132,13 +132,14 @@ func IsWorkflowCompleted(wf *wfv1.Workflow) bool {
 
 // SubmitOpts are workflow submission options
 type SubmitOpts struct {
-	Name           string   // --name
-	GenerateName   string   // --generate-name
-	InstanceID     string   // --instanceid
-	Entrypoint     string   // --entrypoint
-	Parameters     []string // --parameter
-	ParameterFile  string   // --parameter-file
-	ServiceAccount string   // --serviceaccount
+	Name           string                // --name
+	GenerateName   string                // --generate-name
+	InstanceID     string                // --instanceid
+	Entrypoint     string                // --entrypoint
+	Parameters     []string              // --parameter
+	ParameterFile  string                // --parameter-file
+	ServiceAccount string                // --serviceaccount
+	OwnerReference metav1.OwnerReference // useful if your custom controller creates argo workflow resources
```

Comment:
I think it make sense to use pointer instead of struct:

 `*metav1.OwnerReference` 
```
if opts.OwnerReference != nil {
}
```

Reply:
@alexmt sure, makes sense. I have modified to use a pointer.